[New Concept Documents] : comparisons concept #2456 (#2733)
exercism/python: concepts/comparisons/introduction.md

# Introduction

> A [comparison operator][comparisons] in Python (_also called a Python relational operator_), looks at the values of two operands and returns `True` or `False` based on whether the `comparison` condition is met. The most common comparison operators are `"<"`, `">"`, `"=="`, `">="`, `"<="`, and `"!="`. They all have the same priority (which is higher than that of the Boolean operations)

Reviewer: ```suggestion
A [comparison operator][comparisons] in Python (_also called a Python relational operator_), looks at the values of two operands and returns `True` or `False` based on whether the `comparison` condition is met.
The most common comparison operators are `"<"`, `">"`, `"=="`, `">="`, `"<="`, and `"!="`.
They all have the same priority (which is higher than that of the Boolean operations)
```

As per the [build markdown][1] style guide; single sentence per line.

[1]: https://github.com/exercism/docs/blob/main/building/markdown/markdown.md

## Comparison Chaining

> Comparisons can be chained arbitrarily, e.g., `x < y <= z` is equivalent to `x < y` `and` `y <= z`, except that `y` is evaluated only once (but in both cases `z` is _not_ evaluated at all when `x < y` is found to be `False`). This is also called `short-circuit` evaluation which means the execution is stopped if the truth value of the expression has already been determined. Note that the evaluation of expression takes place from left to right. In python, short circuiting is supported by various boolean operators, functions and, in this case, comparison chaining.

Reviewer: ```suggestion
Comparisons can be chained arbitrarily, e.g., `x < y <= z` is equivalent to `x < y` `and` `y <= z`, except that `y` is evaluated only once (but in both cases `z` is _not_ evaluated at all when `x < y` is found to be `False`).
This is also called `short-circuit` evaluation which means the execution is stopped if the truth value of the expression has already been determined.
Note that the evaluation of expression takes place from left to right.
In python, short circuiting is supported by various boolean operators, functions and, in this case, comparison chaining.
```

As per the [build markdown][1] style guide; single sentence per line.

[1]: https://github.com/exercism/docs/blob/main/building/markdown/markdown.md

## Comparison of different data types

> Since everything in Python is an `object`, things can get interesting when objects of different types are compared. For example, the `str` value of a number is considered completely different from the `integer` or `floating-point` value. However, an `integer` **can** be considered equal to a `float`, as they are both numeric types that Python can implicitly convert to compare. For other numeric types, comparison operators are defined where they "make sense", but throw a `TypeError` if the underlying objects cannot be converted for comparison. For more information on the rules that python uses for numeric conversion, see [arithmetic conversions][arithmetic conversions] in the Python documentation.

Reviewer: ```suggestion
Since everything in Python is an `object`, things can get interesting when objects of different types are compared.
For example, the `str` value of a number is considered completely different from the `integer` or `floating-point` value.
However, an `integer` **can** be considered equal to a `float`, as they are both numeric types that Python can implicitly convert to compare.
For other numeric types, comparison operators are defined where they "make sense", but throw a `TypeError` if the underlying objects cannot be converted for comparison.
For more information on the rules that python uses for numeric conversion, see [arithmetic conversions][arithmetic conversions] in the Python documentation.
```

As per the [build markdown][1] style guide; single sentence per line.

[1]: https://github.com/exercism/docs/blob/main/building/markdown/markdown.md

# Introduction

> A comparison operator in python, also called python relational operator, compares the values of two operands and returns `True` or `False` based on whether the condition is met. The most common comparison operators are `"<"`, `">"`, `"=="`, `">="`, `"<="`, `"!="`, `"is"`

Reviewer 2: ```suggestion
A [comparison operator][comparisons] in Python (_also called a Python relational operator_), looks at the values of two operands and returns `True` or `False` based on whether the `comparison` condition is met. The most common comparison operators are `"<"`, `">"`, `"=="`, `">="`, `"<="`, and `"!="`.

```

```python

> >>> 7 > 5
> True
> >>> 99 < 100
> True
> >>> 4 < 4
> False
> >>> 4 <= 4
> True
> >>> 1 >= 1
> True
> >>> 5 == 5
> True
> >>> 6 != 6 # not equal to
> False
> >>> 5 != 6
> True

Reviewer 2: ```suggestion

```

Reviewer 2: The introduction document is displayed to the student _before_ they've completed the exercise.  We try to make them as succinct as we can.  After they complete the exercise, they will be able to see the `about.md` doc.  I think in this circumstance, we  should omit the code examples from the introduction, but have them available to refer back to in the `about.md` after the student has completed the exercise -- if that makes sense?

Author: Makes sense to me. Checked a couple of introductions of other concepts and they don't have code sample either. 
To maintain consistency let's remove them

```

## Comparison Chaining

> Comparisons can be chained arbitrarily, e.g., `x < y <= z` is equivalent to `x < y` `and` `y <= z`, except that `y` is evaluated only once (but in both cases `z` is _not_ evaluated at all when `x < y` is found to be `False`).

Reviewer 2: Besides omitting the code examples, I think I would make the same edits to the prose here as I did in the `about.md`.

Author: Okay sounds good. 

Now

- code blocks from `introduction.md` have been removed
- prose is matched as per `about.md`
